lib/attachments.js

```diff
@@ -4,13 +4,13 @@ const { AttachmentProcessor } = require('@elastic.io/component-commons-library')
 const { unixTimeToIsoDate } = require('./utils/utils');
 const { MAX_FILE_SIZE } = require('./constants');
 
-async function addAttachment(msg, name, stream, contentLength) {
+async function addAttachment(msg, name, getStream, contentLength) {
   try {
     if (contentLength > MAX_FILE_SIZE) {
       throw new Error(`File size is ${contentLength} bytes, it violates the variable MAX_FILE_SIZE, which is currently set to ${MAX_FILE_SIZE} bytes`);
     }
-    const result = await new AttachmentProcessor().uploadAttachment(stream);
-    const attachmentUrl = `${result.config.url}${result.data.objectId}?storage_type=maester`;
+    const attachmentId = await new AttachmentProcessor().uploadAttachment(getStream);
+    const attachmentUrl = new AttachmentProcessor().getMaesterAttachmentUrlById(attachmentId);
```

Comment:
```suggestion
    const attachmentProcessor = new AttachmentProcessor();
    const attachmentId = await attachmentProcessor.uploadAttachment(getStream);
    const attachmentUrl = attachmentProcessor.getMaesterAttachmentUrlById(attachmentId);
```
Usually better create it once (more memory effective use)